src/checker.rs

```diff
@@ -1,87 +1,137 @@
-use anyhow::{anyhow, Result};
+use anyhow::{anyhow, bail, Result};
 use log::warn;
 use std::fmt;
-use std::process::{Command, Output};
+use std::process::Command;
 
-fn check_cmd(program: &str, arg: &str) -> Result<Output> {
-    Command::new(program).arg(arg).output().map_err(Into::into)
+struct BinDep {
+    program: String,
+    installed: bool,
+    version: Option<Version>,
+    required_version: Option<Version>,
+}
+
+impl BinDep {
+    fn build(
+        program: &str,
+        arg: &'static str,
+        version_prefix: Option<&'static str>,
+        required_version: Option<Version>,
+    ) -> Result<Self> {
+        let output = Command::new(program).arg(arg).output()?;
+        let installed = output.status.success();
+        let version = version_prefix
+            .filter(|_| installed)
+            .and_then(|prefix| Version::parse_with_prefix(prefix, output.stdout).ok());
+        Ok(BinDep {
+            program: program.to_string(),
+            installed,
+            version,
+            required_version,
+        })
+    }
+
+    /// Check if the required version is met, return true if no required version
+    fn meet_required_version(&self) -> bool {
+        self.required_version
+            .as_ref()
+            .map(|required_version| {
+                self.version
+                    .as_ref()
+                    .map(|version| version >= required_version)
+                    .unwrap_or(true)
+            })
+            .unwrap_or(true)
+    }
 }
 
 pub struct Checker {
-    cargo: bool,
-    docker: bool,
-    ckb_cli: Option<Vec<u8>>,
+    bin_deps: Vec<BinDep>,
 }
 
 impl Checker {
     pub fn build(ckb_cli_bin: &str) -> Result<Self> {
-        let cargo = check_cmd("cargo", "version")
-            .map(|output| output.status.success())
-            .unwrap_or(false);
-        let docker = check_cmd("docker", "version")
-            .map(|output| output.status.success())
-            .unwrap_or(false);
-        let ckb_cli = check_cmd(ckb_cli_bin, "--version")
-            .map(|output| output.stdout)
-            .ok();
-        Ok(Checker {
-            cargo,
-            docker,
-            ckb_cli,
+        let bin_deps = [
+            ("cargo", "version", None, None),
+            ("docker", "version", None, None),
+            ("cross-util", "--version", None, None),
+            (
+                ckb_cli_bin,
+                "--version",
+                Some("ckb-cli"),
+                Some(REQUIRED_CKB_CLI_VERSION),
+            ),
+        ]
+        .map(|(program, arg, version_prefix, required_version)| {
+            BinDep::build(program, arg, version_prefix, required_version)
         })
+        .into_iter()
+        .collect::<Result<Vec<_>>>()?;
+        Ok(Checker { bin_deps })
     }
 
     pub fn check_ckb_cli(&self) -> Result<()> {
-        if self.ckb_cli.is_none() {
-            return Err(anyhow!("Can't find ckb-cli"));
+        let ckb_cli_dep = &self.bin_deps[3];
```

Comment:
`self.bin_deps[3]` is not very readable/maintainable. We'd need to update it everywhere if the index changes.